framework/src/controller/bus.ts

```diff
@@ -167,10 +167,28 @@ export class Bus {
 	public async invoke<T>(
 		rawRequest: string | JSONRPC.RequestObject,
 	): Promise<JSONRPC.ResponseObjectWithResult<T>> {
-		const request =
-			typeof rawRequest === 'string'
-				? (JSON.parse(rawRequest) as JSONRPC.RequestObject)
-				: rawRequest;
+		let request!: JSONRPC.RequestObject;
+
+		// As the request can be invoked from external source, so we should validate if it exists and valid JSON object
+		if (!rawRequest) {
+			this.logger.error('Empty invoke request.');
+			throw new JSONRPC.JSONRPCError(
+				'Invalid invoke request.',
+				JSONRPC.errorResponse(null, JSONRPC.invalidRequest()),
+			);
+		}
+
+		try {
+			request =
```

Comment:
we had request schema validation in place, we can't use that here to validate the json rpc request?